core/src/main/java/org/apache/iceberg/rest/requests/PlanTableScanRequest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.rest.requests;
+
+import java.util.List;
+import javax.annotation.Nullable;
+import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.rest.RESTRequest;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface PlanTableScanRequest extends RESTRequest {
+
+  @Nullable
+  Long snapshotId();
+
+  @Nullable
+  List<String> select();
```

Comment:
I don't think any of the List types in the request/response classes need to be marked as `@Nullable`. Immutables will use automatically an empty collection in this case if this wasn't set through the builder. I'd suggest to remove all `@Nullable` from the introduced request/response classes where the type is a collection

Reply:
I think @rdblue recommendation for me is not use immutables, so assuming this comment wont apply anymore?